common/flatpak-run.c

```diff
@@ -2961,8 +2961,40 @@ flatpak_run_app (const char     *app_ref,
 
   if (app_deploy != NULL)
     {
+      g_autofree const char **previous_ids = NULL;
+
       app_files = flatpak_deploy_get_files (app_deploy);
 
+      previous_ids = flatpak_deploy_data_get_previous_ids (app_deploy_data);
```

Comment:
we should reverse the list so that we use the most recent ID first, if it exists